source/gui/settingsDialogs.py

```diff
@@ -149,21 +149,13 @@ def __init__(
 		self.makeSettings(self.settingsSizer)
 
 		self.mainSizer.Add(self.settingsSizer, border=guiHelper.BORDER_FOR_DIALOGS, flag=wx.ALL | wx.EXPAND, proportion=1)
-		self.mainSizer.Add(wx.StaticLine(self), flag=wx.EXPAND)
-
-		buttonSizer = guiHelper.ButtonHelper(wx.HORIZONTAL)
-		# Translators: The Ok button on a NVDA dialog. This button will accept any changes and dismiss the dialog.
-		buttonSizer.addButton(self, label=_("OK"), id=wx.ID_OK)
-		# Translators: The cancel button on a NVDA dialog. This button will discard any changes and dismiss the dialog.
-		buttonSizer.addButton(self, label=_("Cancel"), id=wx.ID_CANCEL)
+		buttons = wx.OK | wx.CANCEL

```

Comment:
You are definitely right about translation here.
I took the ID approach because of the current mixture you mention.
Actually, every single time we use a standard message box with Yes/No/Apply/OK/Cancel or even the standard file picker or folder picker, these currently appear in the language of the OS, not the language of NVDA.
It is pretty common practice, and I guess native speakers of languages in which Windows itself is not translated are quite used to this.
To fully circumvent this situation, we would have to:
- Re-implement every single standard message box we emit. A lot of work, but doable.
- Step back from taking advantage of the `wx.Dialog.CreateSeparatedButtonSizer` facility.
- Implement custom file and folder pickers. This one is much more tricky if you want - and should - provide the appropriate look and feel for every single version of Windows NVDA supports.

Thus, IMHO, the easiest and safest path is to use IDs instead of custom translations everywhere applicable, be it only to ensure coherence among the different part of the GUI.